src/tsserver/webServer.ts

```diff
@@ -77,11 +97,26 @@ namespace ts.server {
             // tryEnableSourceMapsForHost?(): void;
             // debugMode?: boolean;
 
-            // Future for semantic server mode
-            fileExists: returnFalse, // Module Resolution
-            directoryExists: returnFalse, // Module resolution
+            // For semantic server mode
+            fileExists: (path: string): boolean => {
+                if (!path.startsWith("http:") && !path.startsWith("https:")) {
+                    return false;
+                }
+
+                const request = new XMLHttpRequest();
+                request.open("HEAD", path, /* asynchronous */ false);
+                request.send();
+
+                return request.status === 200;
+            },
+            directoryExists: (_path: string): boolean => {
+                return false;
+            },
             readDirectory: notImplemented, // Configured project, typing installer
-            getDirectories: () => [], // For automatic type reference directives
+            getDirectories: (path: string) => {
+                console.log("paths", path);
+                return [];
+            }, // For automatic type reference directives

```

Comment:
I am guessing this wasnt intentional
```suggestion
            getDirectories: () => [], // For automatic type reference directives
```